cmd/bosun/sched/alertRunner.go

```diff
@@ -39,15 +48,30 @@ func (s *Schedule) RunAlert(a *conf.Alert) {
 	s.checksRunning.Add(1)
 	// ensure when an alert is done it is removed from the wait group
 	defer s.checksRunning.Done()
+	var lastCheckID int64 = -1
+	// Calcaulate runEvery based on system default and override if an alert has a
+	// custom runEvery
+	runEvery := s.SystemConf.GetDefaultRunEvery()
+	if a.RunEvery != 0 {
+		runEvery = a.RunEvery
+	}
 	for {
-		// Calcaulate runEvery based on system default and override if an alert has a
-		// custom runEvery
-		runEvery := s.SystemConf.GetDefaultRunEvery()
-		if a.RunEvery != 0 {
-			runEvery = a.RunEvery
+		var ctx *checkContext
+		for {
+			checkContextLock.RLock()
+			ctx = s.ctx
+			checkContextLock.RUnlock()
+			//make sure the context has actually changed
+			if ctx.id != lastCheckID {
+				lastCheckID = ctx.id
+				break
+			}
+			collect.Add("check.context_not_changed", opentsdb.TagSet{"alert": a.Name}, 1)
+			time.Sleep(time.Second)
```

Comment:
I initially chose a millisecond. If this occurs at all, it is because the alert and the global context are perfectly lock stepped, and this one the race. A second should be more than adequate to "unlock" it in a timely manner. 

If the counter explodes, we have a bigger problem, and some bad assumptions.